src/commands.ts

```diff
@@ -76,4 +77,48 @@ export function addCommands(
       await settings.set('columnOrder', order);
     }
   });
+  app.commands.addCommand(CommandIDs.showStarred, {
+    isToggleable: true,
+    isToggled: () => {
+      return settings.composite
+        .starredSection as ISettingsLayout['starredSection'];
+    },
+    label: trans.__('Show starred section'),
+    execute: async () => {
+      const starredSection = settings.composite
+        .starredSection as ISettingsLayout['starredSection'];
+      await settings.set('starredSection', !starredSection);
+    },
+    icon: () => {
+      const starredSection = settings.composite
+        .starredSection as ISettingsLayout['starredSection'];
+      if (starredSection) {
+        return checkIcon;
+      } else {
+        return undefined;
+      }
+    }
+  });
```

Comment:
I belive that if instead of using `import { Menu } from '@lumino/widgets';` you do `import { MenuSvg } from '@jupyterlab/ui-components';` then you should be able to remove the icon as it should be automatically derived from `isToggled` state.